xdsl/dialects/arm_neon.py

```diff
@@ -98,6 +103,15 @@ def map_to_num_els(self):
         map = {"D": 2, "S": 4, "H": 8}
         return map[self.name]
 
+    @staticmethod
+    def get_arrangement_from_vec_type(vec_type: VectorType):
+        map_vec_to_arr = {
+            "vector<8xf16>": NeonArrangement.H,
+            "vector<4xf32>": NeonArrangement.S,
+            "vector<2xf64>": NeonArrangement.D,
+        }
+        return map_vec_to_arr[str(vec_type)]
```

Comment:
Can you please make this a map from the instance of the vector type instead, and move the dictionary outside the method so that it's not created every time the function is called? Can you please add tests for supported and unsupported vectors also?

Reply:
I'm not sure I follow. I thought you couldn't use types as keys (unless you implement a separate equals method or something similar)

Reply:
> I'm not sure I follow. I thought you couldn't use types as keys (unless you implement a separate equals method or something similar)

e.g, this was my understanding of what you described, but it doesn't work because of this:
```
def get_arrangement_from_vec_type(vec_type: VectorType):
        map_vec_to_arr = {
            VectorType(Float16Type(), [8]): NeonArrangement.H,
            VectorType(Float32Type(), [4]): NeonArrangement.S,
            VectorType(Float64Type(), [2]): NeonArrangement.D,
        }
        return map_vec_to_arr[vec_type]
```

Reply:
I don't understand this message, what doesn't work in the text above?